File: 2551.put-marbles-in-bags.cs
/*
 * @lc app=leetcode id=2551 lang=csharp
 *
 * [2551] Put Marbles in Bags
 */

// @lc code=start
using System;
using System.Collections.Generic;

public class Solution
{
    /*
    Find k max / min pairs of weights[0] + weights[-1] + (weights[i] + weight[i+1]), i from 0 to n - 2    
    */

    public long PutMarbles(int[] weights, int k)
    {
        if (k == 1)
        {
            return 0;
        }

        List<long> data = new List<long>();
        for (int i = 0; i < weights.Length - 1; i++)
        {
            data.Add(Convert.ToInt64(weights[i] + weights[i + 1]));
        }

        data.Sort();
        return data.GetRange(data.Count - k + 1, k - 1).Sum() - data.GetRange(0, k - 1).Sum();
    }
}

// @lc code=end

File: 2551.put-marbles-in-bags.py
#
# @lc app=leetcode id=2551 lang=python3
#
# [2551] Put Marbles in Bags
#

# @lc code=start
class Solution:
    '''
    Find k max / min pairs of weights[0] + weights[-1] + (weights[i] + weight[i+1]), i from 0 to n - 2    
    '''

    def putMarbles(self, weights: List[int], k: int) -> int:
        if k == 1:
            return 0
        data = []
        for i in range(len(weights) - 1):
            data.append(weights[i] + weights[i+1])
        data.sort()
        return sum(data[(-k + 1):]) - sum(data[:(k - 1)])
        
# @lc code=end

